spec/features/users_spec.rb

```diff
@@ -0,0 +1,74 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :feature do
+  before do
+    @user1 = User.create!(name: 'Tom', posts_counter: 0)
+    @user2 = User.create!(name: 'Toro', posts_counter: 0, bio: 'developer')
+    @user3 = User.create!(name: 'Ted', posts_counter: 0)
+
+    visit users_path
+  end
+
+  describe 'index page' do
+    it 'displays usernames of all other users' do
+      expect(page).to have_content('Tom')
+      expect(page).to have_content('Toro')
+      expect(page).to have_content('Ted')
+    end
+
+    it 'displays profile picture for each user' do
+      expect(page).to have_css('div.user-photo', count: 5)
+    end
+
+    it 'displays the number of posts each user has written' do
+      expect(page).to have_content('Number of posts: 0', count: 3) # Assuming each user has 5 posts
+    end
+
+    it 'redirects to user show page when clicking on a user' do
```

Comment:
- [ ] Please as a better coding practice separate the test for the user index and user show into separate files 👍🏾